Reconstruction/RecInterface/RecInterface/ITowerTool.h

```diff
@@ -0,0 +1,67 @@
+#ifndef RECINTERFACE_ITOWERTOOL_H
+#define RECINTERFACE_ITOWERTOOL_H
+
+// Gaudi
+#include "GaudiKernel/IAlgTool.h"
+
+// datamodel
+#include "datamodel/CaloCluster.h"
+
+/** @class ITowerTool RecInterface/RecInterface/ITowerTool.h ITowerTool.h
+ *
+ *  Abstract interface to tower building tool.
+ *
+ *  @author Anna Zaborowska
+ */
+
+class ITowerTool: virtual public IAlgTool {
+public:
+  DeclareInterfaceID(ITowerTool, 1, 0);
+  /**  Find number of calorimeter towers.
+   *   @return Array containing number of towers in eta and phi.
+   */
+  virtual std::array<int,2> towersNumber() = 0;
+  /**  Find number of calorimeter towers in eta for current event.
+   *   @return Number of towers in eta for current event.
+   */
+  virtual int etaTowersNumber() = 0;
+  /**  Build calorimeter towers.
+   *   @param[out] aTowers Calorimeter towers.
+   *   @return Size of the cell collection.
+   */
+  virtual uint buildTowers(std::vector<std::vector<float>> & aTowers) = 0;
+  /**  Find cells belonging to a cluster.
+   *   @param[in] aEta Position of the middle tower of a cluster in eta
+   *   @param[in] aPhi Position of the middle tower of a cluster in phi
+   *   @param[in] aHalfEtaFinal Half size of cluster in eta (in units of tower size). Cluster size is 2*aHalfEtaFinal+1
+   *   @param[in] aHalfPhiFinal Half size of cluster in phi (in units of tower size). Cluster size is 2*aHalfPhiFinal+1
+   *   @param[out] aEdmCluster Cluster where cells are attached to
+   */
+  virtual void matchCells(float, float, uint, uint, fcc::CaloCluster& aEdmCluster) = 0;
```

Comment:
Can you add the respective names to the first arguments (or stick to the conventions of unnamed arguments in the declaration)?